### ISSUE TITLE
Calling autoUpdater.checkForUpdates() multiple times causes existing download to be wiped

### ISSUE THREAD

**Issue by mobitar** [electron-userland/electron-builder#2006]
```
"electron": "1.6.11"
"electron-builder": "^15.0.0"
"electron-updater": "^1.8.1"
```

(I've updated all the above versions after posting this issue; not sure if updating fixes it yet)

I call `autoUpdater.checkForUpdates()` when the app window appears.

I noticed sometimes the Mac app doesn't update, even after the update is downloaded (no issues in the logs). I found that after the `update-downloaded` event is received, if I call `autoUpdater.checkForUpdates()` again, then **quit the app before the download is finished**, then the auto-update will not be installed.

I think calling checkForUpdates() will begin downloading the update, even if it has already been downloaded before. Based on the observed behavior, it looks like it deletes any currently downloaded updates immediately.

Instead, I think it should delete the old update only after the new one has finished downloading.

Can you confirm this behavior?

**Comment by develar** [comment]
Already fixed in the latest versions. Please upgrade.

**Comment by develar** [comment]
Duplicates #1788

**Comment by mobitar** [comment]
@develar unfortunately I am still seeing this issue with electron-builder 19.45.5 and electron 1.7.9.

I'm not sure if the underlying cause is the same, but on the surface, here's what happens:

1. Open Mac app, receive `update-downloaded` event.
2. Instruct user to quit app and re-open to install update.
3. User quits app, and re-opens immediately. 

Result:
- App is not upgraded. `update-downloaded` is received again. Repeat steps 1-3, same results.

After repeating many times, it will eventually update.

Any ideas as what could be causing this? Does user have to wait X seconds after quitting app before re-opening?

**Comment by develar** [comment]
@mobitar your code looks correct. I am not aware of such issue. Please file your comment as a new issue to be not forgotten.

**Comment by cyphercodes96** [comment]
Windows NSIS same issue

Dev Dependencies:

   "electron": "^4.2.9",
    "electron-builder": "^21.2.0",
    "electron-download": "^4.1.1",
    "electron-publisher-s3": "^20.15.0",
    "electron-reload": "1.4.0",


Any resolutions?
